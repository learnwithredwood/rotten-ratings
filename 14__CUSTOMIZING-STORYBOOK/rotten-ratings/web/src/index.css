@import url('https://fonts.googleapis.com/css2?family=Abril+Fatface&family=Inter:wght@300;400;500;700&display=swap');

/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */

 @layer base {
  body {
    @apply text-white;
  }
 }
 @layer components { }
 @layer utilities { }
