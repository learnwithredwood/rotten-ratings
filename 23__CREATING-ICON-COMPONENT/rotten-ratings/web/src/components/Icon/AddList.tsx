interface AddListProps {
  size?: number
}

const AddList = ({ size = 24 }: AddListProps) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M6.72273 7.89093C7.12189 7.89093 7.44547 7.56735 7.44547 7.1682C7.44547 6.76904 7.12189 6.44547 6.72273 6.44547C6.32358 6.44547 6 6.76904 6 7.1682C6 7.56735 6.32358 7.89093 6.72273 7.89093Z" />
      <path d="M19.0094 6.44547H8.89111V7.89093H19.0094V6.44547Z" />
      <path d="M6.72273 10.7819C7.12189 10.7819 7.44547 10.4583 7.44547 10.0591C7.44547 9.65997 7.12189 9.3364 6.72273 9.3364C6.32358 9.3364 6 9.65997 6 10.0591C6 10.4583 6.32358 10.7819 6.72273 10.7819Z" />
      <path d="M19.0094 9.3364H8.89111V10.7819H19.0094V9.3364Z" />
      <path d="M6.72273 13.6728C7.12189 13.6728 7.44547 13.3492 7.44547 12.9501C7.44547 12.5509 7.12189 12.2273 6.72273 12.2273C6.32358 12.2273 6 12.5509 6 12.9501C6 13.3492 6.32358 13.6728 6.72273 13.6728Z" />
      <path d="M19.0094 12.2273H8.89111V13.6728H19.0094V12.2273Z" />
      <path d="M6.72273 16.5637C7.12189 16.5637 7.44547 16.2401 7.44547 15.841C7.44547 15.4418 7.12189 15.1183 6.72273 15.1183C6.32358 15.1183 6 15.4418 6 15.841C6 16.2401 6.32358 16.5637 6.72273 16.5637Z" />
      <path d="M16.1184 15.1183H8.89111V16.5637H16.1184V15.1183Z" />
      <path d="M23.3458 17.2865H21.1776V15.1183H19.7322V17.2865H17.564V18.7319H19.7322V20.9001H21.1776V18.7319H23.3458V17.2865Z" />
    </svg>
  )
}

export { AddList }
