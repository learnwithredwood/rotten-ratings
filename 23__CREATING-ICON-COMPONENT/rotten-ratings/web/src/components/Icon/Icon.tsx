interface IconProps {
  // Props
}

const Icon = ({}: IconProps) => {
  return (
    <div>
      <h2>{'Icon'}</h2>
      <p>{'Find me in ./web/src/components/Icon/Icon.tsx'}</p>
    </div>
  )
}

export { Icon, IconProps }
