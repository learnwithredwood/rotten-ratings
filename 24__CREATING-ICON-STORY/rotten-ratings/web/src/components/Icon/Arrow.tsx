interface ArrowProps {
  size?: number
}

const Arrow = ({ size = 24 }: ArrowProps) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>Arrow Icon</title>
      <path d="M10.0666 7.40068V24H13.9333V7.40068L18.3663 11.8337L21.1 9.10005L12 0L2.8999 9.10005L5.63359 11.8337L10.0666 7.40068Z" />
    </svg>
  )
}

export { Arrow }
