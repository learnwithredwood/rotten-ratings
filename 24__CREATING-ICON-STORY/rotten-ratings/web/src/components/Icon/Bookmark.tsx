interface BookmarkProps {
  size?: number
}

const Bookmark = ({ size = 24 }: BookmarkProps) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>Bookmark Icon</title>
      <path d="M16.111 19.276C16.0388 19.3482 15.9521 19.4069 15.8541 19.4478C15.5592 19.5703 15.2178 19.5021 14.9917 19.276L12.1933 16.4776L9.395 19.276C9.1689 19.5021 8.8275 19.5703 8.53256 19.4478C8.23705 19.3246 8.04397 19.0364 8.04397 18.7163L8.04397 5.28428C8.04397 4.84662 8.39768 4.49291 8.83533 4.49291L15.5513 4.49291C15.989 4.49291 16.3427 4.84662 16.3427 5.28428L16.3427 18.7163C16.3427 18.9301 16.2565 19.1304 16.111 19.276ZM9.6267 16.8056L11.6337 14.7986C11.9432 14.4891 12.4435 14.4891 12.753 14.7986L14.76 16.8056V6.07565L9.6267 6.07565L9.6267 16.8056Z" />
    </svg>
  )
}

export { Bookmark }
