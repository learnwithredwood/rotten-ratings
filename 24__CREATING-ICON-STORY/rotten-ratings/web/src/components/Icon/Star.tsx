interface StarProps {
  size?: number
}

const Star = ({ size = 24 }: StarProps) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>Star Icon</title>
      <path d="M12 17.27L18.18 21L16.54 13.97L22 9.24L14.81 8.63L12 2L9.19 8.63L2 9.24L7.46 13.97L5.82 21L12 17.27Z" />
    </svg>
  )
}

export { Star }
