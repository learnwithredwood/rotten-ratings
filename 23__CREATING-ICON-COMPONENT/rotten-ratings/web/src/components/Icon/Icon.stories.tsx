// See https://storybook.js.org/docs/react/writing-stories/args.

import type { ComponentMeta } from '@storybook/react'

import { Icon, type IconProps } from './Icon'

export default {
  title: 'Components/Icon',
  component: Icon,
} as ComponentMeta<typeof Icon>

const Template = (args) => <Icon {...args} />

export const Primary = Template.bind({})
Primary.args = {
  // args
}

