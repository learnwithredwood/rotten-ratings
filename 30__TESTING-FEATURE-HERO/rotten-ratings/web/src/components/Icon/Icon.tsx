import { AddList } from './AddList'
import { Arrow } from './Arrow'
import { Bookmark } from './Bookmark'
import { Close } from './Close'
import { Hide } from './Hide'
import { Search } from './Search'
import { Show } from './Show'
import { Star } from './Star'
import { WatchLater } from './WatchLater'

interface IconProps {
  name: string
  size?: number
}

const Icon = ({ name, size = 24 }: IconProps) => {
  switch (name.toLowerCase()) {
    case 'addlist':
      return <AddList size={size} />
    case 'arrow':
      return <Arrow size={size} />
    case 'bookmark':
      return <Bookmark size={size} />
    case 'close':
      return <Close size={size} />
    case 'hide':
      return <Hide size={size} />
    case 'search':
      return <Search size={size} />
    case 'show':
      return <Show size={size} />
    case 'star':
      return <Star size={size} />
    case 'watchlater':
      return <WatchLater size={size} />
    default:
      return <div />
  }
}

export { Icon, IconProps }
