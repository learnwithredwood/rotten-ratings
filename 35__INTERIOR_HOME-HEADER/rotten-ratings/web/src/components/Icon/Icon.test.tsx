import { render, screen } from '@redwoodjs/testing/web'

import { Icon } from './Icon'

//   Improve this test with help from the Redwood Testing Doc:
//    https://redwoodjs.com/docs/testing#testing-components

describe('Icon', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<Icon name="AddList" />)
    }).not.toThrow()
  })

  // correct icon
  it('renders to AddList icon', () => {
    render(<Icon name="AddList" />)
    const icon = screen.getByTestId('AddListIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Arrow icon', () => {
    render(<Icon name="Arrow" />)
    const icon = screen.getByTestId('ArrowIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Bookmark icon', () => {
    render(<Icon name="Bookmark" />)
    const icon = screen.getByTestId('BookmarkIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Close icon', () => {
    render(<Icon name="Close" />)
    const icon = screen.getByTestId('CloseIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Hamburger icon', () => {
    render(<Icon name="Hamburger" />)
    const icon = screen.getByTestId('HamburgerIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Hide icon', () => {
    render(<Icon name="Hide" />)
    const icon = screen.getByTestId('HideIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Search icon', () => {
    render(<Icon name="Search" />)
    const icon = screen.getByTestId('SearchIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Show icon', () => {
    render(<Icon name="Show" />)
    const icon = screen.getByTestId('ShowIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to Star icon', () => {
    render(<Icon name="Star" />)
    const icon = screen.getByTestId('StarIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  it('renders to WaterLater icon', () => {
    render(<Icon name="WatchLater" />)
    const icon = screen.getByTestId('WatchLaterIcon')
    expect(icon).toBeInTheDocument()
    expect(icon).toMatchSnapshot()
  })

  // size
  it('renders to the correct size', () => {
    render(<Icon name="AddList" size={64} />)
    const icon = screen.getByTestId('AddListIcon')
    expect(icon).toHaveAttribute('width', '64')
    expect(icon).toHaveAttribute('height', '64')
  })
})
