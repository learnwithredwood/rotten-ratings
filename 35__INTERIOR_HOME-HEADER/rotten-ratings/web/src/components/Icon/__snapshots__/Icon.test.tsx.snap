// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders to AddList icon 1`] = `
<svg
  data-testid="AddListIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Add List Icon
  </title>
  <path
    d="M6.72273 7.89093C7.12189 7.89093 7.44547 7.56735 7.44547 7.1682C7.44547 6.76904 7.12189 6.44547 6.72273 6.44547C6.32358 6.44547 6 6.76904 6 7.1682C6 7.56735 6.32358 7.89093 6.72273 7.89093Z"
  />
  <path
    d="M19.0094 6.44547H8.89111V7.89093H19.0094V6.44547Z"
  />
  <path
    d="M6.72273 10.7819C7.12189 10.7819 7.44547 10.4583 7.44547 10.0591C7.44547 9.65997 7.12189 9.3364 6.72273 9.3364C6.32358 9.3364 6 9.65997 6 10.0591C6 10.4583 6.32358 10.7819 6.72273 10.7819Z"
  />
  <path
    d="M19.0094 9.3364H8.89111V10.7819H19.0094V9.3364Z"
  />
  <path
    d="M6.72273 13.6728C7.12189 13.6728 7.44547 13.3492 7.44547 12.9501C7.44547 12.5509 7.12189 12.2273 6.72273 12.2273C6.32358 12.2273 6 12.5509 6 12.9501C6 13.3492 6.32358 13.6728 6.72273 13.6728Z"
  />
  <path
    d="M19.0094 12.2273H8.89111V13.6728H19.0094V12.2273Z"
  />
  <path
    d="M6.72273 16.5637C7.12189 16.5637 7.44547 16.2401 7.44547 15.841C7.44547 15.4418 7.12189 15.1183 6.72273 15.1183C6.32358 15.1183 6 15.4418 6 15.841C6 16.2401 6.32358 16.5637 6.72273 16.5637Z"
  />
  <path
    d="M16.1184 15.1183H8.89111V16.5637H16.1184V15.1183Z"
  />
  <path
    d="M23.3458 17.2865H21.1776V15.1183H19.7322V17.2865H17.564V18.7319H19.7322V20.9001H21.1776V18.7319H23.3458V17.2865Z"
  />
</svg>
`;

exports[`Icon renders to Arrow icon 1`] = `
<svg
  data-testid="ArrowIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M10.0666 7.40068V24H13.9333V7.40068L18.3663 11.8337L21.1 9.10005L12 0L2.8999 9.10005L5.63359 11.8337L10.0666 7.40068Z"
  />
</svg>
`;

exports[`Icon renders to Bookmark icon 1`] = `
<svg
  data-testid="BookmarkIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Bookmark Icon
  </title>
  <path
    d="M16.111 19.276C16.0388 19.3482 15.9521 19.4069 15.8541 19.4478C15.5592 19.5703 15.2178 19.5021 14.9917 19.276L12.1933 16.4776L9.395 19.276C9.1689 19.5021 8.8275 19.5703 8.53256 19.4478C8.23705 19.3246 8.04397 19.0364 8.04397 18.7163L8.04397 5.28428C8.04397 4.84662 8.39768 4.49291 8.83533 4.49291L15.5513 4.49291C15.989 4.49291 16.3427 4.84662 16.3427 5.28428L16.3427 18.7163C16.3427 18.9301 16.2565 19.1304 16.111 19.276ZM9.6267 16.8056L11.6337 14.7986C11.9432 14.4891 12.4435 14.4891 12.753 14.7986L14.76 16.8056V6.07565L9.6267 6.07565L9.6267 16.8056Z"
  />
</svg>
`;

exports[`Icon renders to Close icon 1`] = `
<svg
  data-testid="CloseIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Close Icon
  </title>
  <path
    d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
  />
</svg>
`;

exports[`Icon renders to Hamburger icon 1`] = `
<svg
  data-testid="HamburgerIcon"
  fill="none"
  height="30"
  viewBox="0 0 42 30"
  width="42"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M41.9569 2H0V0H41.9569V2Z"
  />
  <path
    d="M41.9569 16H0V14H41.9569V16Z"
  />
  <path
    d="M41.9569 30H0V28H41.9569V30Z"
  />
</svg>
`;

exports[`Icon renders to Hide icon 1`] = `
<svg
  data-testid="HideIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Hide
  </title>
  <path
    d="M12 7C14.76 7 17 9.24 17 12C17 12.65 16.87 13.26 16.64 13.83L19.56 16.75C21.07 15.49 22.26 13.86 22.99 12C21.26 7.61 16.99 4.5 11.99 4.5C10.59 4.5 9.25 4.75 8.01 5.2L10.17 7.36C10.74 7.13 11.35 7 12 7ZM2 4.27L4.28 6.55L4.74 7.01C3.08 8.3 1.78 10.02 1 12C2.73 16.39 7 19.5 12 19.5C13.55 19.5 15.03 19.2 16.38 18.66L16.8 19.08L19.73 22L21 20.73L3.27 3L2 4.27ZM7.53 9.8L9.08 11.35C9.03 11.56 9 11.78 9 12C9 13.66 10.34 15 12 15C12.22 15 12.44 14.97 12.65 14.92L14.2 16.47C13.53 16.8 12.79 17 12 17C9.24 17 7 14.76 7 12C7 11.21 7.2 10.47 7.53 9.8ZM11.84 9.02L14.99 12.17L15.01 12.01C15.01 10.35 13.67 9.01 12.01 9.01L11.84 9.02Z"
  />
</svg>
`;

exports[`Icon renders to Search icon 1`] = `
<svg
  data-testid="SearchIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Search Icon
  </title>
  <path
    d="M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z"
  />
</svg>
`;

exports[`Icon renders to Show icon 1`] = `
<svg
  data-testid="ShowIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Search Icon
  </title>
  <path
    d="M12 4.5C7 4.5 2.73 7.61 1 12C2.73 16.39 7 19.5 12 19.5C17 19.5 21.27 16.39 23 12C21.27 7.61 17 4.5 12 4.5ZM12 17C9.24 17 7 14.76 7 12C7 9.24 9.24 7 12 7C14.76 7 17 9.24 17 12C17 14.76 14.76 17 12 17ZM12 9C10.34 9 9 10.34 9 12C9 13.66 10.34 15 12 15C13.66 15 15 13.66 15 12C15 10.34 13.66 9 12 9Z"
  />
</svg>
`;

exports[`Icon renders to Star icon 1`] = `
<svg
  data-testid="StarIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Star Icon
  </title>
  <path
    d="M12 17.27L18.18 21L16.54 13.97L22 9.24L14.81 8.63L12 2L9.19 8.63L2 9.24L7.46 13.97L5.82 21L12 17.27Z"
  />
</svg>
`;

exports[`Icon renders to WaterLater icon 1`] = `
<svg
  data-testid="WatchLaterIcon"
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Watch Later Icon
  </title>
  <path
    d="M10.834 8.4243V13.2425H14.0461V11.6364H12.4401V8.4243H10.834Z"
  />
  <path
    d="M12.4415 3.60608C14.63 3.60608 16.6501 4.48059 18.1341 5.9654L19.6963 4.40911V9.22733H14.8597L16.9951 7.10009C15.8077 5.91159 14.1923 5.21215 12.4415 5.21215C8.8858 5.21215 5.99274 8.09425 5.99274 11.6364C5.99274 15.1786 8.8858 18.0607 12.4415 18.0607V19.6668C7.99669 19.6668 4.38056 16.0652 4.38056 11.6364C4.38056 7.2077 7.99669 3.60608 12.4415 3.60608Z"
  />
</svg>
`;
